misc-utils/rename.c

```diff
@@ -53,20 +54,31 @@ static int string_replace(char *from, char *to, char *s, char *orig, char **newn
 	return 0;
 }
 
-static int ask(char *name)
+static int ask(char *name, int nooverwrite)
 {
-	int c = 'n', r;
+	char buf[BUFSIZ];
+	int x;
+
 	printf(_("Overwrite `%s'? "), name);
```

Comment:
This patch means that expected reply from user is: Yes, No and  \n. The \n is defaults to nooverwrite flag. It would be nice make it obvious to the user and somehow describe this in the prompt
```
Overwrite `%s'  " [Y]es/[N]o (enter defaults to %s): ", name, nooverwrite ? _("no overwrite") : _("overwrite")
```
Frankly, not sure if I like this complicated semantic. It seems like over-engineering. Maybe it would be better to ignore --no-overwrite option at all if --interactive is specified and expect Yes or No.

Reply:
Note that in main() you can mark --no-overwrite and --interactive mutually exclusive and keep the code simple ;-)